### ISSUE TITLE
[feature request] weekly.seasonality.prior.scale and yearly.seasonality.prior.scale

### ISSUE THREAD

**Issue by wjhrdy** [facebookincubator/prophet#58]
Is there a way to break out the weights for the seasonality components so that you can vary the weight of weekly seasonality separately from yearly seasonality?

I find that often I'm very confident about the weekly seasonality and not super confident about yearly so I'd like to lower the effect of yearly seasonality and keep weekly seasonality at the same weight.

Adding these arguments would be a solution.

weekly.seasonality.prior.scale 

yearly.seasonality.prior.scale

**Comment by seanjtaylor** [comment]
I wonder if it might make sense to make more generic, like `prior_scales = list()`.

**Comment by wjhrdy** [comment]
If there is a reason that we would want to expand that list of prior scales in the future past weekly seasonality, yearly seasonality, changepoint and holiday it might be good to make it a list.

One scenario I can think of is if you have a large list of holiday components and you want to weight them differently you may want to list those weights in a more general prior_scales list.

One drawback of using this prior_scales list is that it's a less standard input and if you are tailoring this to be easy to use and automatic this added complexity of arguments may be prohibitively complex.

**Comment by peacej** [comment]
Definitely agree, would like to only decrease yearly.seasonality.prior.scale but not the weekly one.

**Comment by bletham** [comment]
Would anyone like to make a PR for this in the v0.2 branch? It would need to be on top of the `add_regressor` feature (https://github.com/facebookincubator/prophet/commit/8f1607cd93d4641d690d95c5f4dd2737e8e7152f), which makes it more convenient to have prior scales with different values. This has been done for Python but is not yet in the R version (see #101).

* For holidays, I think we could add an optional column to the dataframe `prior_scale` (optional like `lower_window` and `upper_window`) which specifies the prior_scale for each holiday if given, and otherwise it defaults to holidays_prior_scale. There should be validation that a single prior scale is specified for each holiday. The `make_holiday_features` method could return the prior scales in a list, which could then immediately be used by `make_all_seasonality_features` as-is.

* Custom seasonalities (via `add_seasonality`) should be able to specify the prior scale, but then default to seasonality_prior_scale if not provided.

* There are three built-in seasonalities: yearly, weekly, and daily. Of the options discussed above, the cleanest seems to me to be having three corresponding arguments to Prophet(), yearly_seasonality_prior_scale, yearly_seasonality_prior_scale, and yearly_seasonality_prior_scale. I do worry about cluttering it with too many arguments, but also wouldn't want to sacrifice clarity and ease-of-use over two arguments. Any other thoughts on this?

**Comment by bletham** [comment]
This is done in Py with https://github.com/facebookincubator/prophet/commit/a620a6c9f9ddddc0e9a43d38254796c259669c6f and https://github.com/facebookincubator/prophet/commit/8d276433391022a53010c378b3afa2e263716528 and can be ported to R once #278 is finished. I'll add documentation for it then.

**Comment by bletham** [comment]
This can now be done in v0.2 available in CRAN and pypi. Documentation is here:
https://facebookincubator.github.io/prophet/docs/seasonality_and_holiday_effects.html

If you have any issues with it feel free to reopen here, or to make a new issue.
